chia/consensus/block_body_validation.py

```diff
@@ -436,20 +426,24 @@ async def validate_block_body(
             return Err.WRONG_PUZZLE_HASH, None
 
     # 21. Verify conditions
-    for npc in npc_list:
-        assert height is not None
-        unspent = removal_coin_records[npc.coin_name]
-        error = mempool_check_conditions_dict(
-            unspent,
-            npc.condition_dict,
+    # verify absolute/relative height/time conditions
+    if npc_result is not None:
+        assert npc_result.conds
```

Comment:
`assert npc_result.conds is not None`

I avoid `assert X` unless X is a bool, because it is highly confusion for an optional of an empty list.